tsl/src/nodes/data_node_dispatch.c

```diff
@@ -1063,8 +1077,14 @@ get_insert_attrs(Relation rel)
 	{
 		Form_pg_attribute attr = TupleDescAttr(tupdesc, i);
 
-		if (!attr->attisdropped)
-			attrs = lappend_int(attrs, AttrOffsetGetAttrNumber(i));
+		if (attr->attisdropped
+#if PG12_GE
+			|| attr->attgenerated != '\0'
```

Comment:
Nit: Even  a check for `attr->attgenerated` should work instead of comparing with `'\0'`

Reply:
I am not sure I follow what you are asking/suggesting.

`attgenerated` is a char that map to different types of generated columns. Currently, only `ATTRIBUTE_GENERATED_STORED` is supported/exists. I assume that there will be more types of generated columns in the future, so I figured it is better to check for the '\0' case in order to be forward compatible with new `ATTRIBUTE_GENERATED_` in the future.